File: program/program/reverse-words-in-a-string/reverse_words_in_a_string.pl
#!/usr/bin/perl

use strict;
use warnings;

sub reverse_words {
    my ($input_string) = @_;

    # Split the input string into words
    my @words = split /\s+/, $input_string;

    # Reverse the order of words
    my $output_string = join ' ', reverse @words;

    return $output_string;
}

# Example usage
my $input_string = "Hello World";
my $output_string = reverse_words($input_string);

print "Input  : \"$input_string\"\n";
print "Output : \"$output_string\"\n";

File: program/program/reverse-words-in-a-string/reverse_words_in_a_string.go
package main

import (
	"fmt"
	"strings"
)

// reverseWords reverses the order of words in a given string.
func reverseWords(input string) string {
	// Split the input string into words.
	words := strings.Fields(input)

	// Reverse the order of words using a two-pointer approach.
	for i, j := 0, len(words)-1; i < j; i, j = i+1, j-1 {
		words[i], words[j] = words[j], words[i]
	}

	// Join the reversed words into a string and return.
	return strings.Join(words, " ")
}

func main() {
	// Example usage of reverseWords function.
	inputString := "Hello World"
	reversedResult := reverseWords(inputString)
	fmt.Println(reversedResult)
}
